registry/storage/garbagecollect.go

```diff
@@ -47,6 +54,31 @@ func MarkAndSweep(ctx context.Context, storageDriver driver.StorageDriver, regis
 		}
 
 		err = manifestEnumerator.Enumerate(ctx, func(dgst digest.Digest) error {
+			if opts.RemoveUntagged {
+				tags, err := repository.Tags(ctx).Lookup(ctx, distribution.Descriptor{Digest: dgst})
+				if err != nil {
+					return fmt.Errorf("failed to retrieve tags for digest %v: %v", dgst, err)
+				}
+				if len(tags) == 0 {
+					_, err := pathFor(manifestRevisionPathSpec{name: repoName, revision: dgst})
```

Comment:
Sorry, should have been more clear before. What is the purpose of running this function and not using the result? `pathFor` only formats a path for a given resource.